lib/Parse/ParseDecl.cpp

```diff
@@ -3776,7 +3777,7 @@ std::vector<Decl *> Parser::parseDeclListDelayed(IterableDeclContext *IDC) {
     return { };
   }
 
-  auto BeginParserPosition = getParserPosition({BodyRange.Start,BodyRange.End});
+  auto BeginParserPosition = getParserPosition({BodyRange.Start, SourceLoc()});
```

Comment:
What is this change about?

Reply:
`getParserPosition()` receives `PersistentParserState::ParserPos` which receives the token location and its *previous* location. That loc is to be restored to `PreviousLoc` in parser. So `BodyLoc.End` was just a mistake.

I admit that this change is not needed to this PR though.